HW06.md

```diff
@@ -0,0 +1,1131 @@
+    ## 
+    ##     select
+
+    ## Loading required package: AnnotationFilter
+
+    ## 
+    ## Attaching package: 'AnnotationFilter'
+
+    ## The following object is masked from 'package:magrittr':
+    ## 
+    ##     not
+
+    ## 
+    ## Attaching package: 'ensembldb'
+
+    ## The following object is masked from 'package:dplyr':
+    ## 
+    ##     filter
+
+    ## The following object is masked from 'package:stats':
+    ## 
+    ##     filter
+
+``` r
+library(EnsDb.Hsapiens.v75)
+
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+#Because I combined my data long-wise, many values would be removed with the following code (because my gene names show up more than once). Robert told me that it's okay to not use the following functions and still be able to do my analysis. 
+
+#ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+```
+
+Do the t-test and make a table of the t-test results\!
+
+``` r
+#detach functions gave me an error, so I just specified which library some of my functions are from  
+  #detach(package:EnsDb.Hsapiens.v75, unload = TRUE)
+  #detach(package:ensembldb, unload = TRUE)
+library(broom)
+
+#added CDC42 as my own gene of interest
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "CDC42")
+
+
+GBM.genes.of.interest <- dplyr::filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#save table so I don't have to rerun all my previous code to do my analysis
+#write_csv(GBM.genes.of.interest, "GBM.genes.of.interest.csv")
+
+GBM.genes.of.interest <- read.csv("GBM.genes.of.interest.csv")
+
+
+#creating function to filter by gene and then group by type
+grouped_genes <- function(x){
+  GBM.genes.of.interest %>%  
+  dplyr::filter(gene_symbol == x) %>%
+    group_by(type)
+}
+
+Group_genes <- lapply(genes_of_interest, grouped_genes)
+
+#creating function to conduct two-sample t-test
+ttest <- function(x){
+  t.test(TPM~type, data = x) }
+  
+ttest_values <- lapply(Group_genes,ttest)
+
+#relabel so it is clear what gene the statistics belong to
+for (x in 1:length(ttest_values))  {
+ttest_values[[x]]$data.name <-genes_of_interest[[x]]
+ }
+
+#print out the t-test results
+print(ttest_values)
```

Comment:
I really liked the output of the tidy() function in the broom package for visualizing a large number of t-test results. I could see it as especially useful if you want to filter for genes that had a certain result, like a p-value below 0.01. I think you could be able to implement it into your code by piping the result of line 624 into the tidy() function. You would then have to use rbind() to make a single data frame.